controller/src/main/java/org/jboss/as/controller/operations/common/ProcessEnvironment.java

```diff
@@ -117,6 +127,26 @@ protected abstract boolean isRuntimeSystemPropertyUpdateAllowed(String propertyN
      */
     protected abstract void systemPropertyUpdated(String propertyName, String propertyValue);
 
+    /**
+     * Get the UUID of this process.
+     * @return the UUID of this process.
+     */
+    public abstract UUID getInstanceUuid();
+
+    protected UUID createProcessUUID(final Path filePath) throws IOException {
+        UUID uuid = null;
+        if (Files.exists(filePath)) {
+            String line = Files.lines(filePath).findFirst().get();
+            uuid = UUID.fromString(line);
+        }
+        if (uuid == null) {
+            uuid = UUID.randomUUID();
+            Files.createDirectories(filePath.getParent());
+            Files.write(filePath, Collections.singletonList(uuid.toString()), StandardOpenOption.CREATE);
```

Comment:
Actually I should get @dmlloyd opinion on that. It is probably okay not to use a privileged block.